### ISSUE TITLE
post_run method fails to add exceptions to job log

### ISSUE THREAD

**Issue by dnewood** [nautobot/nautobot#1807]
### Environment
* Python version: 3.7.13
* Nautobot version: 1.3.2

### Steps to Reproduce
1. Create a job with a `post_run()` method that should raise an Exception. Example:
```python
class TestPostRun(Job):

    class Meta:
        name = "Test Post Run"
        description = "Testing Post Run Stuffz"
        commit_default = True

    def run (self, data, commit):
        self.log_success(message="Yay, Job ran!")
    
    def post_run(self):
        self.log_success(message="Post run has started! Let's break stuff.")
        
        # Purposely cause an exception
        lol_fake_object.add(idk_this_should_break)
        
        self.log_info(message="Testing to see if stuff is broken.")
```
2. Run the job.

### Expected Behavior
Expected to see a log entry within the job showing that the `post_run` group experienced a NameError exception.

### Observed Behavior
The post_run method just "eats" the exception and stops the job silently. You can however see the expected exception in the worker logs:
```
[2022-05-24 22:06:00,915: ERROR/ForkPoolWorker-2] Task nautobot.extras.jobs.run_job[a9df438c-d93c-47ca-8771-a1b7f0af9fa6] raised unexpected: NameError("name 'lol_fake_object' is not defined")
Traceback (most recent call last):
  File "/usr/local/lib/python3.7/site-packages/celery/app/trace.py", line 451, in trace_task
    R = retval = fun(*args, **kwargs)
  File "/usr/local/lib/python3.7/site-packages/celery/app/trace.py", line 734, in __protected_call__
    return self.run(*args, **kwargs)
  File "/usr/local/lib/python3.7/site-packages/nautobot/extras/jobs.py", line 1185, in run_job
    _run_job()
  File "/usr/local/lib/python3.7/site-packages/nautobot/extras/jobs.py", line 1172, in _run_job
    output = job.post_run()
  File "/opt/nautobot/jobs/CreateFwPolicyv2.py", line 463, in post_run
    lol_fake_object.add(idk_this_should_break)
NameError: name 'lol_fake_object' is not defined
```
However the Job in the GUI makes it seem like everything went fine:
![Screen Shot 2022-05-24 at 6 06 42 PM](https://user-images.githubusercontent.com/67761754/170143376-65261572-1c27-481c-912b-90a38a3ec106.png)

**Comment by jathanism** [comment]
Briefly: `post_run()` is called outside of the atomic block, so it means any exceptions raised therein don't bubble up to Nautobot and instead die in silence on the worker process. This results in inconsistent state between the `JobResult` object and the underlying Celery task state. 

See:

https://github.com/nautobot/nautobot/blob/d0f868c9f436caed9a12bdf41205d7f613ecbc3a/nautobot/extras/jobs.py#L1138-L1171

- Atomic block exits on line 1161
- `post_run()` called on line 1171

**Comment by glennmatthews** [comment]
Not sure why this didn't auto-close. Fixed in #2334.
